### ISSUE TITLE
Failed to create symlink over an existing folder even with `-force`

### ISSUE THREAD

**Issue by Flamefire** [plougher/squashfs-tools#282]
TLDR: `unsquash` fails to overwrite a folder with a symlink

I have a squashfs image file that has a couple of symlinks. After extraction to an empty directory (`/foo/root`) it looks like:

```
# ll /foo/root/var/
insgesamt 16
drwxrwxr-x  4 root     root     4096 Mai 23  2014 ./
drwxr-xr-x 18 root     root     4096 Mär  8  2021 ../
lrwxrwxrwx  1 root     root        6 Feb 27  2014 cache -> ../tmp/
drwxrwxr-x  2 root     root     4096 Feb 27  2014 lib/
lrwxrwxrwx  1 root     root        6 Feb 27  2014 lock -> ../tmp/
lrwxrwxrwx  1 root     root        6 Feb 27  2014 log -> ../tmp/
lrwxrwxrwx  1 root     root        6 Feb 27  2014 pcmcia -> ../tmp/
lrwxrwxrwx  1 root     root        6 Feb 27  2014 run -> ../tmp/
lrwxrwxrwx  1 root     root        6 Feb 27  2014 spool -> ../tmp/
lrwxrwxrwx  1 root     root        6 Feb 27  2014 tmp -> ../tmp/
```

Now in a generic program some directories are pre-created, one of them is `/var/tmp`. E.g. assume `mkdir /foo/root && mkdir -p /foo/root/var/tmp`.

Unpacking the same image to that destination with `-force` fails with an error:
> create_inode: failed to create symlink /foo/root/var/tmp, because File exists

I think the issue is that only files or (sym)links (but no folders) are removed at
https://github.com/plougher/squashfs-tools/blob/5a3bf7f08415fe865df326614faa95531a1b94c8/squashfs-tools/unsquashfs.c#L1238-L1239

**Comment by Flamefire** [comment]
I tested this with singularity (where I found this to cause a blocking issue) and can confirm that the fix works.
